### PR TITLE
Recreate EMA to avoid `InferenceTensor` error during NaN recovery

### PR DESCRIPTION
**Pull request by Y-T-G**
**MRE**
```bash
yolo train model=yolo11n.pt data=coco128.yaml epochs=10 exist_ok lr0=0.05 optimizer=Adam
```

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Documentation polish and training stability improvements: clearer “Coming Soon” labels, refined SAM 2/3 docs, nav tweaks, and a fix for EMA state loading when resuming training—plus a more conservative NaN recovery trigger. ✨

### 📊 Key Changes
- Docs wording consistency:
  - Standardized “⚠️ Coming Soon” (was “COMING SOON”) for YOLO26 and SAM3 across docs. 📝
- SAM pages updates:
  - SAM 2: removed outdated “SAM 2.1” tip; streamlined headings. 🎯
  - SAM 3: improved keywords, heading placement, grammar (“What is…”), and updated citation (ICLR 2026 submission with OpenReview link and paper ID). 🔗
- Docs navigation:
  - Reordered models in mkdocs to place YOLO26 after YOLO12 for clearer progression. 🧭
- Trainer fixes:
  - When loading checkpoints, reinitialize EMA with `ModelEMA(self.model)` before loading EMA weights to avoid inference-tensor update errors. 🛠️
  - Tightened NaN recovery condition: now triggers only if loss is NaN/Inf and fitness is NaN/Inf or collapsed. 🧪

### 🎯 Purpose & Impact
- Better clarity and consistency in docs:
  - Users see uniform “Coming Soon” messaging and up-to-date SAM 2/3 info, including a direct reference to the SAM 3 submission status. 📚
- Smoother training resumes:
  - Fix prevents errors when resuming training with EMA checkpoints after validation, reducing interruptions for users using EMA. ✅
- Reduced false positives in recovery:
  - More conservative NaN recovery avoids unnecessary training resets, improving training stability—though it may detect purely fitness-based issues less aggressively. ⚖️

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Y-T-G, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! This is an automated response — an Ultralytics engineer will also review and assist soon. To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes (especially around NaN recovery and checkpoint resume), and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the minimum necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

🐛 Since this PR appears to address a bug related to NaN recovery during checkpoint loading, please provide a Minimum Reproducible Example (MRE) if not already included:
- A short script or clear steps to reproduce the NaN recovery error
- Environment details (OS, Python, PyTorch, CUDA)
- Relevant logs or stack traces
- The command(s) you ran and any custom settings used

Example MRE template (feel free to adapt):
```bash
# commands used
yolo train model=... data=... epochs=... resume=True
```
```python
# minimal snippet triggering the issue
from ultralytics import YOLO
model = YOLO("...pt")
# simulate/trigger NaN -> resume/load path
model.train(data="...", epochs=1, resume=True)
```

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don’t hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Comment by Y-T-G** [comment]
@glenn-jocher @Laughing-q I don't know if we should have this enabled for fitness collapse. Because it just enters a loop of recovery and collapse, which makes sense because the training is deterministic and it will lead to the same collapse every time.

`yolo train model=yolo11n.pt data=coco128.yaml epochs=10 exist_ok lr0=0.05 optimizer=Adam`

```
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, method='weighted_average', num_output_channels=3), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       1/10      4.57G      2.091      3.314      1.972        120        640: 100% ━━━━━━━━━━━━ 8/8 2.5it/s 3.2s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 4/4 2.3it/s 1.7s
                   all        128        929      0.211    0.00112   2.55e-05   6.56e-06

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       2/10      5.59G      3.142      4.973      2.992         68        640: 100% ━━━━━━━━━━━━ 8/8 10.3it/s 0.8s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 4/4 0.6it/s 6.2s
                   all        128        929          0          0          0          0
WARNING ⚠️ Fitness collapse detected (attempt 1/3), recovering from last.pt...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       2/10      5.59G      2.954      4.869      2.865         96        640: 100% ━━━━━━━━━━━━ 8/8 14.0it/s 0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 4/4 2.6it/s 1.5s
                   all        128        929      0.211    0.00117   2.17e-05    5.1e-06

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       3/10       5.6G      3.612  1.838e+04      4.249        124        640: 100% ━━━━━━━━━━━━ 8/8 11.3it/s 0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 4/4 1.3it/s 3.1s
                   all        128        929          0          0          0          0
WARNING ⚠️ Fitness collapse detected (attempt 1/3), recovering from last.pt...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       3/10       5.6G      2.999      4.944      2.864         70        640: 100% ━━━━━━━━━━━━ 8/8 12.7it/s 0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 4/4 2.6it/s 1.5s
                   all        128        929      0.211    0.00117   2.17e-05    5.1e-06

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       4/10       5.6G      3.697  2.094e+05        525        145        640: 100% ━━━━━━━━━━━━ 8/8 10.9it/s 0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 4/4 1.6it/s 2.5s
                   all        128        929          0          0          0          0
WARNING ⚠️ Fitness collapse detected (attempt 1/3), recovering from last.pt...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       4/10       5.6G       3.04      5.013      2.907        111        640: 100% ━━━━━━━━━━━━ 8/8 12.1it/s 0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 4/4 2.6it/s 1.5s
                   all        128        929      0.211    0.00117   2.17e-05    5.1e-06

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       5/10       5.6G       3.93  2.832e+05       1449        160        640: 100% ━━━━━━━━━━━━ 8/8 11.7it/s 0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 4/4 1.6it/s 2.6s
                   all        128        929          0          0          0          0
WARNING ⚠️ Fitness collapse detected (attempt 1/3), recovering from last.pt...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       5/10      5.61G      3.016      4.863      2.903         94        640: 100% ━━━━━━━━━━━━ 8/8 10.8it/s 0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 4/4 2.6it/s 1.5s
                   all        128        929      0.211    0.00117   2.17e-05    5.1e-06

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       6/10      5.62G      3.706  2.758e+06       9694        133        640: 100% ━━━━━━━━━━━━ 8/8 11.9it/s 0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 4/4 2.1it/s 1.9s
                   all        128        929          0          0          0          0
WARNING ⚠️ Fitness collapse detected (attempt 1/3), recovering from last.pt...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       6/10      5.63G      2.995      4.879      2.903         87        640: 100% ━━━━━━━━━━━━ 8/8 11.6it/s 0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 4/4 2.6it/s 1.5s
                   all        128        929      0.211    0.00117   2.17e-05    5.1e-06

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       7/10      5.63G      3.882  9.422e+05       3755        126        640: 100% ━━━━━━━━━━━━ 8/8 10.5it/s 0.8s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 4/4 1.3it/s 3.0s
                   all        128        929          0          0          0          0
WARNING ⚠️ Fitness collapse detected (attempt 1/3), recovering from last.pt...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       7/10      5.63G      3.008       4.85       2.89        106        640: 100% ━━━━━━━━━━━━ 8/8 12.1it/s 0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 4/4 2.6it/s 1.5s
                   all        128        929      0.211    0.00117   2.17e-05    5.1e-06

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       8/10      5.63G      3.659  1.968e+06  1.822e+04        105        640: 100% ━━━━━━━━━━━━ 8/8 11.7it/s 0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 4/4 1.7it/s 2.3s
                   all        128        929          0          0          0          0
WARNING ⚠️ Fitness collapse detected (attempt 1/3), recovering from last.pt...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       8/10      5.63G      3.024      5.004      2.892        108        640: 100% ━━━━━━━━━━━━ 8/8 11.4it/s 0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 4/4 2.6it/s 1.6s
                   all        128        929      0.211    0.00117   2.17e-05    5.1e-06

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       9/10      5.63G      3.592  4.762e+05      588.4        136        640: 100% ━━━━━━━━━━━━ 8/8 11.0it/s 0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 4/4 1.4it/s 2.9s
                   all        128        929          0          0          0          0
WARNING ⚠️ Fitness collapse detected (attempt 1/3), recovering from last.pt...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       9/10      5.63G      3.009      5.021      2.892         88        640: 100% ━━━━━━━━━━━━ 8/8 12.1it/s 0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 4/4 2.6it/s 1.5s
                   all        128        929      0.211    0.00117   2.17e-05    5.1e-06

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      10/10      5.63G      3.236      156.9      3.182        134        640: 100% ━━━━━━━━━━━━ 8/8 12.1it/s 0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 4/4 0.5it/s 7.3s
                   all        128        929          0          0          0          0
WARNING ⚠️ Fitness collapse detected (attempt 1/3), recovering from last.pt...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      10/10      5.63G      2.986      4.958      2.898         82        640: 100% ━━━━━━━━━━━━ 8/8 12.2it/s 0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 4/4 2.6it/s 1.5s
                   all        128        929      0.211    0.00117   2.17e-05    5.1e-06

10 epochs completed in 0.019 hours.
Optimizer stripped from /ultralytics/runs/detect/train/weights/last.pt, 5.5MB
Optimizer stripped from /ultralytics/runs/detect/train/weights/best.pt, 5.5MB
```

**Comment by glenn-jocher** [comment]
@Y-T-G yeah this looks bad, should only act on NaN, not on 0 fitness.

I thought the RNG wasn't reset though so new random numbers would be used for the retry.

**Comment by glenn-jocher** [comment]
@Y-T-G https://github.com/ultralytics/ultralytics/pull/22372/commits/ebae4f25b35a3145e3f2b477c9a22510d5a24951 should ignore fitness collapse without an accompanied nan loss. Can you test after this change?

**Comment by UltralyticsAssistant** [comment]
🎉 Huge thanks for the merge, @Y-T-G — and kudos to @glenn-jocher for the contributions!

“Well done is better than well said.” — Benjamin Franklin. This PR embodies that spirit: polished docs with clear “⚠️ Coming Soon” for YOLO26 and refined SAM 2/3 pages, improved navigation flow (placing YOLO26 after YOLO12), and meaningful training stability upgrades with robust EMA checkpoint handling and a more conservative NaN recovery. The result is clearer guidance for users and smoother, more resilient training resumes.

Appreciate the attention to detail and the steady hand on reliability. This is the kind of craftsmanship that keeps our community moving forward. Thank you! 🙌

**Comment by Y-T-G** [comment]
@glenn-jocher Yes. It doesn't run recovery anymore with 0 fitness.

**Comment by glenn-jocher** [comment]
Seems to still be working incorrectly, it's running on inf loss when metrics are good (see Slack).
